src/lib/index/index_service.cpp

```diff
@@ -89,7 +89,7 @@ void IndexService::start() {
     }
     // Force taking cp after recovery done. This makes sure that the index table is in consistent state and dirty buffer
     // after recovery can be added to dirty list for flushing in the new cp
-    hs()->cp_mgr().trigger_cp_flush(true /* force */);
+    hs()->cp_mgr().trigger_cp_flush(true /* force */).wait();
```

Comment:
trigger_cp_flush(true) waits until cp flush is done. Did you test it without .wait()

Reply:
Right now all the tests are done without the wait()... I dont know if we do need a wait() but it seems make things easier...